custom_components/airthings_wave/airthings.py

```diff
@@ -192,152 +206,159 @@ def __init__(self, scan_interval, mac=None):
                             sensor_data = None
                             if str(characteristic.uuid) in sensor_decoders:
-                                char = self._dev.getCharacteristics(uuid=characteristic.uuid)[0]
-                                data = char.read()
+                                data = await self._dev.read_gatt_char(characteristic.uuid)
                                 sensor_data = sensor_decoders[str(characteristic.uuid)].decode_data(data)
                                 _LOGGER.debug("{} Got sensordata {}".format(mac, sensor_data))
-                                
+                            
+                            # ToDo: Is there a more elegant way to handle the below?
+
                             if str(characteristic.uuid) in command_decoders:
-                                self.delgate.data = None # Clear the delegate so it is ready for new data.
-                                char = self._dev.getCharacteristics(uuid=characteristic.uuid)[0]
-                                # Do these steps to get notification to work, I do not know how it works, this link should explain it
-                                # https://devzone.nordicsemi.com/guides/short-range-guides/b/bluetooth-low-energy/posts/ble-characteristics-a-beginners-tutorial
-                                desc, = char.getDescriptors(forUUID=CHAR_UUID_CCCD)
-                                desc.write(struct.pack('<H', 1), True)
-                                char.write(command_decoders[str(characteristic.uuid)].cmd)
-                                for i in range(3):
-                                    if self._dev.waitForNotifications(0.1):
-                                        _LOGGER.debug("Received notification, total data received len {}".format(len(self.delgate.data)))
-                                
-                                sensor_data = command_decoders[str(characteristic.uuid)].decode_data(self.delgate.data)
-                                _LOGGER.debug("{} Got cmddata {}".format(mac, sensor_data))
+                                _LOGGER.debug("command characteristic: {}".format(characteristic.uuid))
+                                # Set up the notification handlers
+                                await self._dev.start_notify(characteristic.uuid, self.notification_handler)
+                                # send command to this 'indicate' characteristic
+                                await self._dev.write_gatt_char(characteristic.uuid, command_decoders[str(characteristic.uuid)].cmd)
+                                # Wait for up to one second to see if a callblack comes in.
+                                for i in range(10):
+                                    await asyncio.sleep(0.1)
+                                    # Check if there is data
+                                    _LOGGER.debug("waiting for callback: {}".format(i))
+                                    if self._command_data != None:
```

Comment:
Need to use `is not None` instead of `!= None`

Reply:
I fixed this, thanks for the feedback